nemo/collections/multimodal/data/clip/mbeir_dataset.py

```diff
@@ -0,0 +1,487 @@
+                        neg_cand_id_list[i % len(neg_cand_id_list)]
+                    )  # % Wrap around from idx 0.
+                for neg_cand_did in selected_neg_cand_id_list:
+                    neg_cand = self.cand_pool.get(neg_cand_did, None)
+                    neg_cand_txt = neg_cand.get("txt") or ""
+                    neg_cand_txt = format_string(neg_cand_txt)
+                    neg_cand["txt"] = neg_cand_txt
+                    selected_neg_cand_list.append(neg_cand)
+
+        def _prepare_data_dict(txt, img_path):
+            img = self._load_and_preprocess_image(img_path)
+#             img = img_path
+            return {"txt": txt, "img": img}
+
+        query = _prepare_data_dict(
+            query_txt_with_prompt if self.enable_query_instruct else query_txt_without_prompt, query_img_path
+        )
+        instance = {"query": query}
+
+        if self.mode == Mode.EVAL:
+            if self.returns.get("hashed_qid"):
+                instance.update({"qid": hash_qid(qid)})
+            if self.returns.get("task_id"):
+                instance.update({"task_id": get_mbeir_task_id(query_modality, pos_cand_modality)})
+            # TODO: add src_content if needed
+
+        if self.mode == Mode.TRAIN:
+            if self.returns.get("hashed_p_did"):
+                instance.update({"p_did": hash_did(selected_pos_cand_did)})
+
+            pos_cand = _prepare_data_dict(
+                pos_cand_txt,
+                pos_cand.get("img_path", None),
+            )
+            instance.update({"pos_cand": pos_cand})
+
+            neg_cand_list = [
+                _prepare_data_dict(
+                    neg_cand["txt"],
+                    neg_cand.get("img_path", None),
+                )
+                for neg_cand in selected_neg_cand_list
+            ]
+            if len(neg_cand_list) > 0:
+                instance.update({"neg_cand_list": neg_cand_list})
+        return instance
+
+
+class MBEIRCandidatePoolDataset(MBEIRDatasetBase):
+    def __init__(
+        self,
+        mbeir_data_dir,  # Root directory of the MBEIR dataset
+        cand_pool_data_path,  # Relate path to the candidate pool data
+        img_preprocess_fn,
+        returns=None,  # Catch any return-related settings
+        print_config=True,  # Whether to print the dataset config
+    ):
+        super().__init__(mbeir_data_dir, img_preprocess_fn)
+        self._load_cand_pool(cand_pool_data_path)
+
+        returns = {} if returns is None else returns
+        self.returns = {
+            "src_content": False,  # default value
+            "hashed_did": True,  # default value for candidate id
+            **returns,
+        }
+
+        # Print dataset config
+        if print_config:
+            self.cand_pool_path = cand_pool_data_path
+            self._print_config()
+
+    def _print_config(self):
+        # Print dataset config
+        print(f"\n---Mbeir Candidate Pool Dataset Config---")
```

Comment:
replace print with logging, all nemo models use that and we avoid using print (sorry for nit picking :) )